networking_cisco/plugins/cisco/device_manager/plugging_drivers/aci_vlan_trunking_driver.py

```diff
@@ -0,0 +1,255 @@
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import re
+
+from oslo_config import cfg
+from oslo_log import log as logging
+
+from neutron.common import constants as l3_constants
+from neutron import manager
+from neutron.plugins.common import constants as svc_constants
+
+from networking_cisco.plugins.cisco.device_manager.plugging_drivers import (
+    hw_vlan_trunking_driver as hw_vlan)
+
+LOG = logging.getLogger(__name__)
+
+APIC_OWNED = 'apic_owned_'
+UUID_REGEX = '[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}'
+DEVICE_OWNER_ROUTER_GW = l3_constants.DEVICE_OWNER_ROUTER_GW
+DEVICE_OWNER_ROUTER_INTF = l3_constants.DEVICE_OWNER_ROUTER_INTF
+
+ACI_ASR1K_DRIVER_OPTS = [
+    cfg.StrOpt('aci_transit_nets_config_file', default=None,
+               help=_("ACI with ASR transit network configuration file.")),
+]
+
+cfg.CONF.register_opts(ACI_ASR1K_DRIVER_OPTS, "general")
+
+DEFAULT_EXT_DICT = {'gateway': '1.103.2.1',
+                    'next_hop': '1.103.2.254',
+                    'host_pool_cidr': '1.103.2.0/24',
+                    'host_cidr_ver': 4}
+
+
+class AciVLANTrunkingPlugDriver(hw_vlan.HwVLANTrunkingPlugDriver):
+    """Driver class for Cisco ACI-based devices.
+
+    The driver works with VLAN segmented Neutron networks. It
+    determines which workflow is active (GBP or Neutron), and
+    uses that implementation to get the information needed for
+    the networks between the hosting device and the ACI fabric.
+    """
+    # once initialized _device_network_interface_map is dictionary
+    _device_network_interface_map = None
+    _apic_driver = None
+    _l3_plugin = None
+    _get_vrf_context = None
+
+    def __init__(self):
+        super(AciVLANTrunkingPlugDriver, self).__init__()
+        self._cfg_file = cfg.CONF.general.aci_transit_nets_config_file
+        self._get_ext_net_name = None
+        self._default_ext_dict = DEFAULT_EXT_DICT
+        self._transit_nets_cfg = {}
+
+    @property
+    def transit_nets_cfg(self):
+        if self._cfg_file:
+            networks_dict = open(self._cfg_file, 'r').read()
+            self._transit_nets_cfg = eval(networks_dict)
+        else:
+            self._transit_nets_cfg = {}
+        return self._transit_nets_cfg
+
+    @property
+    def get_ext_net_name(self):
+        # FIXME(tbachman): sadly, only exists to enable UT
+        if self.apic_driver:
+            return self._get_ext_net_name
+        else:
+            return None
+
+    @property
+    def get_vrf_context(self):
+        if self.apic_driver:
+            return self._get_vrf_context
+        else:
+            return None
+
+    def _get_vrf_context_gbp(self, context, router_id, port_db):
```

Comment:
this function looks incomplete, might put a TBD or something? I assume this will only work for ml2 now.